### DISCUSSION TITLE
Initialize array with default values and length

### DISCUSSION THREAD

**Discussion by Tastaturtaste** [slint-ui/slint#2976] [Q&A] [answered]
I have a component with a variable and/or large number of elements. I also have an array with one entry for each component. How can I initialize the array without writing the initial default value for every entry in the array?
Currently:
```
in-out property <int> N: 9;
// false repeated 9 times
// does not resize when N changes
private property <[bool]> on: [false, false, false, false, false, false, false, false, false]; 
```
What I would like:
```
in-out property <int> N: 9;
// resizes automatically when N changes
private property <[bool]> on: [false; N];
```

**Comment by ogoffart** [comment]
This is i think currently not possible to do in .slint code.
You can do that by having the property initialized from native code.

**Reply by Tastaturtaste** [reply]
That would require usage of the `init` callback, which the documentation discourages for the use of property initialization. It's also much less clear for a reader. 

Are there plans to allow array sizes to depend on other properties in the future?

**Reply by ogoffart** [reply, marked as answer]
You don't need to use the init callback. (and indeed shouldn't use it for this kind of things)

You can also do something like this:

```
global Logic {
  pure callback create_array(int) -> [bool];
}

// later
in-out property <int> N: 9;
private property <[bool]> on: Logic.create_array(N);
```

**Reply by Tastaturtaste** [reply]
Thanks, I didn't think about pure callbacks in the global scope.
